internal/scheduler/nodedb/nodedb.go

```diff
@@ -190,7 +197,15 @@ func (nodeDb *NodeDb) CreateAndInsertWithJobDbJobsWithTxn(txn *memdb.Txn, jobs [
 		return err
 	}
 	for _, job := range jobs {
-		if err := bindJobToNodeInPlace(nodeDb.priorityClasses, job, entry); err != nil {
+		priority, ok := job.GetScheduledAtPriority()
+		if !ok {
+			priorityClass, err := interfaces.PriorityClassFromLegacySchedulerJob(nodeDb.priorityClasses, job)
```

Comment:
You're guaranteed to have a jobdb.Job here.

Reply:
I have a weak preference for using the `LegacySchedulerJob` interface here, since that's what's used in all other methods of this type.